ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -101,8 +107,18 @@ public void start() throws StoreException {
               "Failed to acquire lock on file " + dataDir + ". Another process or thread is using this directory.",
               StoreErrorCodes.Initialization_Error);
         }
+        File storeDescriptorFile = new File(dataDir, STORE_DESCRIPTOR);
+        StoreDescriptor storeDescriptor = null;
+        if (storeDescriptorFile.exists()) {
```

Comment:
Looks to me like all of this (lines 110 - 118) can be done inside `StoreDescriptor`. Is there a reason not to? If we are anyway using an abstraction, it looks to me like it would be natural to let the abstraction handle persistence also.